README.md

```diff
@@ -184,6 +188,29 @@ Sometimes, multi-valued fields may contain missing values; in this case, the `wi
 ### Auxiliary files
 `vembrane` supports additional files, such as lists of genes or ids with the `--aux NAME=path/to/file` option. The file should contain one item per line and is parsed as a set. For example `vembrane filter --aux genes=genes.txt "ANN['SYMBOL'] in AUX['genes']" variants.vcf` will keep only records where the annotated symbol is in the set specified in `genes.txt`.
 
+### Ontologies
+`vembrane` supports ontologies in OBO format. The ontology is loaded into memory and can be accessed in the filter expression via the `SO` symbol. This enables filtering based on relationships between ontology terms. 
+For example, `vembrane filter --ontology so.obo 'ANN["Consequence"].any_is_a("intron_variant")'` will keep only records where at least one of the consequences is an intron variant *or a subtype thereof*.
+If no ontology is provided, the built-in ontology from sequenceontology.org (date: 2024-06-06) is loaded automatically if the `SO` symbol is accessed.
+
+There are three relevant classes/types:
+- `Term`: Represents a term in the ontology. It inherits from `str` and can be used as such.
+- `Consequences`: Represents a list of terms. It inherits from `list` and can be used as such.
+- `SO`: Represents the ontology itself. It is a singleton and can be used to access the ontology.
+
+The following functions are available for ontologies, where `term` is a single `Term` and `terms` is a `Consequences` object:
+- `SO.get_id(term: Term) -> str`: Convert from term name (e.g. `stop_gained`) to accession (e.g. `SO:0001587`).
+- `SO.get_term(id_: str) -> Term`: Convert from accession (e.g. `SO:0001587`) to term name (e.g. `stop_gained`).
+- `terms.most_specific_terms() -> Consequences`: Narrow down the list of terms to the most specific ones, e.g. `frameshift_variant&frameshift_truncation&intron_variant&splice_site_variant&splice_donor_5th_base_variant` will lead to `frameshift_truncation&splice_donor_5th_base_variant`.
+- `term.ancestors() -> Consequences`: Get *all* parents of a term.
+- `term.descendants() -> Consequences`: Get *all* children of a term.
+- `term.parents() -> Consequences`: Get parents of a term.
+- `term.children() -> Consequences`: Get children of a term.
+- `term.is_a(parent: Term) -> bool`: Check if there is a path from `term` to `parent`, i.e. whether `term` is the `parent` type or a subtype of it.
+- `terms.any_is_a(parent: str) -> bool`: Check if any of the terms is a subtype of `parent`.
```

Comment:
For is_a, parent is a Term, for any_is_a, it is a str. Intentional?

Reply:
Good catch, these should both be Terms. (Though Term inherits from str anyways, so…)